src/applications/vass/pages/EnterOTC.unit.spec.jsx

```diff
@@ -18,60 +33,207 @@ const defaultRenderOptions = {
 
-    expect(screen.getByTestId('header')).to.exist;
+describe('VASS Component: EnterOTC', () => {
+  beforeEach(() => {
+    mockFetch();
   });
 
-  it('should render success alert with verification message', () => {
-    const { container } = renderWithStoreAndRouterV6(
-      <EnterOTC />,
-      defaultRenderOptions,
-    );
-    const alert = container.querySelector('va-alert[status="success"]');
-
-    expect(alert).to.exist;
-    expect(alert.getAttribute('visible')).to.exist;
+  afterEach(() => {
+    resetFetch();
   });
 
-  it('should display email address in alert message', () => {
-    const { getByText } = renderWithStoreAndRouterV6(
-      <EnterOTC />,
-      defaultRenderOptions,
-    );
-
-    expect(getByText(/We just emailed a one-time verification code to/i)).to
-      .exist;
-    expect(getByText(/test@test.com/i)).to.exist;
-  });
+  it('should initialize correctly', () => {
+    const screen = renderComponent();
 
-  it('should render OTC input field', () => {
-    const { container } = renderWithStoreAndRouterV6(
-      <EnterOTC />,
-      defaultRenderOptions,
+    expect(screen.getByTestId('header')).to.exist;
+    expect(screen.getByTestId('enter-otc-success-alert')).to.exist;
+    expect(screen.getByTestId('enter-otc-success-alert').textContent).to.match(
+      /t\*\*\*@test.com/i,
     );
-    const otcInput = container.querySelector('va-text-input[name="otc"]');
-
+    expect(screen.queryByTestId('enter-otc-error-alert')).to.not.exist;
+    const otcInput = screen.getByTestId('otc-input');
     expect(otcInput).to.exist;
-    expect(otcInput.getAttribute('label')).to.equal(
-      'Enter your one-time verification code',
+    expect(otcInput.getAttribute('label')).to.match(
+      /Enter your one-time verification code/i,
     );
     expect(otcInput.getAttribute('required')).to.exist;
+    expect(screen.getByTestId('continue-button')).to.exist;
+    expect(screen.getByTestId('continue-button').getAttribute('text')).to.match(
+      /Continue/i,
+    );
   });
 
-  it('should render continue button', () => {
-    const { container } = renderWithStoreAndRouterV6(
-      <EnterOTC />,
-      defaultRenderOptions,
-    );
-    const continueButton = container.querySelector('va-button');
+  describe('form validation', () => {
+    it('should show field error when submitting with empty code', async () => {
+      const { getByTestId } = renderComponent();
+      const continueButton = getByTestId('continue-button');
+
+      continueButton.click();
+
+      await waitFor(() => {
+        const otcInput = getByTestId('otc-input');
+        expect(otcInput.getAttribute('error')).to.match(
+          /Please enter your one-time verification code/i,
+        );
+      });
+    });
+  });
+
+  describe('API error handling', () => {
+    it('should display error alert for invalid_otc with multiple attempts remaining', async () => {
+      setFetchJSONFailure(global.fetch.onCall(0), {
+        errors: [
```

Comment:
Maybe we should create fixtures for these so we can use them here and in the mock responses when running locally, or create a ticket to do that.